src/zope/tales/testing.py

```diff
@@ -0,0 +1,12 @@
+import six
+import unittest
+
+
+class TestCase(unittest.TestCase):
+    """Base test case for Python version compatibility."""
+
+    if six.PY2:  # pragma: no cover
+        # Avoid DeprecationWarning for assertRaisesRegexp on Python 3 while
+        # coping with Python 2 not having the Regex spelling variant
+        assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
+                                    unittest.TestCase.assertRaisesRegexp)
```

Comment:
This fancy `getattr()` spelling was lovingly crafted to work without a `six.PY2` guard, IIRC.

If you use a guard, you can just say `if six.PY2: assertRaisesRegex = unittest.TestCase.assertRaisesRegexp`.

Reply:
Good idea! My intent was to make this Python 2 compatibility code easier to discover once we drop Python 2 support.